Automattic/wp-calypso
# G Suite: Remove statuses shown for Google Workspace mailboxes pending ToS acceptance

This pull request, together with D73732-code, removes the statuses shown for Google Workspace mailboxes pending ToS acceptance. As a reminder, a user who purchased a Google Workspace account has to agree to two different types of terms of service: the ones for the account and the ones for the mailbox itself. We were using this last bit of information to prompt users to log into their account. Unfortunately, this information is unreliable and we had several cases where it would be wrong. Since we cannot trust it, this pull request makes the following changes:

* It no longer displays the `Action required` status on the `Domains` page when a Google account has mailboxes pending ToS acceptance. This was confusing because in this context this status should be referring to the domain itself, and not any email solution that depends on it. Moreover, this status would also be displayed at the top of the `Domain Settings` page without additional information, meaning the user had no idea how to actually fix that state.
* It updates the `Email Plan` page so it no longer show mailboxes with a `Pending` status (and its red border) when the account is pending ToS acceptance. I wasn't planning to make that change originally but keeping this status seemed much harder to achieve (because of how our logic is implemented) rather than just removing it. We still display a clear warning and CTA in the header of this page so this shouldn't be a problem though.
* It updates the `Email Plan` page to no longer show an `Action required` status (and its red border) as well as a `Finish setup` button for mailboxes pending ToS acceptance. As mentioned above, this information is unreliable and could be confusing.
* Finally, this pull request removes the components that would display a modal box, and offer users to reset their password. This modal box was no longer accessible anyway, so it didn't make sense to keep that code around. If we want to offer the option to reset a password again, we will rather add it to the `Email Plan` page.


##### `Domains` page

Before | After
------ | -----
<img width="623" alt="DOMAINS - BEFORE" src="https://user-images.githubusercontent.com/594356/151146974-438c170e-59a7-497b-a19e-e5eb8093b481.png"> | <img width="624" alt="DOMAINS - AFTER" src="https://user-images.githubusercontent.com/594356/151147050-39fad2f3-36e4-42cc-b752-e33e7751e091.png">


##### `Email Plan` page for account pending ToS acceptance

Before | After
------ | -----
<img width="624" alt="EMAIL PLAN GS - BEFORE" src="https://user-images.githubusercontent.com/594356/151147159-d5294841-c0db-4617-b243-0821226c5940.png"> | <img width="623" alt="EMAIL PLAN GS - AFTER" src="https://user-images.githubusercontent.com/594356/151147229-41d81b80-b5c4-4c2a-b393-cb3ef2a1894e.png">


##### `Email Plan` page for mailbox pending ToS acceptance

Before | After
------ | -----
<img width="623" alt="EMAIL PLAN GW - BEFORE" src="https://user-images.githubusercontent.com/594356/151147311-6f8bc604-4051-4b75-a823-1e299150914e.png">| <img width="623" alt="EMAIL PLAN GW - AFTER" src="https://user-images.githubusercontent.com/594356/151147414-e50ceac3-2bcd-4f39-9157-a09af631a683.png">


#### Testing instructions

> Note you should apply D73732-code to your sandbox if it has not been merged yet (and point `public-api.wordpress.com` to your sandbox in that case).

1. Run `git checkout update/google-workspace-warnings` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/60460#issuecomment-1021395461)
2. Open the [`Emails` page](http://calypso.localhost:3000/email)
3. Purchase a Google Workspace account, and wait until it is provisioned
4. Navigate to the [`Domains` page](http://calypso.localhost:3000/domains/manage)
5. Assert that the status is `Active` for the domain with this account
6. Head back to the `Emails` page
7. Assert that the status is `Action required` for your new account
8. Click that account to access the `Email Plan` page
9. Assert that there is warning with a `Finish setup` button in the header
10. Assert that there is no button nor status in the list of mailboxes
11. Click the `Finish setup` button
12. Log into your Google Admin Console, and accept both ToS
13. Refresh the `Email Plan` page
14. Assert that the warning in the header has disappeared
